### PR TITLE
ci-trigger: Add support for launching physical tests

### PR DESCRIPTION
**Pull request by bstoll**
This PR adds frontend/UI support for launching physical hardware tests from PR comments.  Note that the "all" keyword does not include physical devices for now - I want physical tests to be explicitly launched while ironing out any problems.

### CODE REVIEW

File: tools/ci-trigger/pr.go [hunk 1]
```diff
@@ -151,6 +158,41 @@ func (p *pullRequest) createBuild(ctx context.Context, buildClient *cloudbuild.S
...
+				for _, v := range physicalDevice.Tests {
+					if v.Status != "pending authorization" {
+						continue physicalDeviceLoop
+					}
+				}
+				jobID, err := uuid.NewRandom()
+				if err != nil {
+					return fmt.Errorf("uuid.NewRandom device %q: %w", physicalDevice.Type.String(), err)
+				}
+				p.Physical[i].ArchivePath = objPath
```
Comment (greg-dennis) [inline tools/ci-trigger/pr.go:176]: isn't p.Physical[i] the same as physicalDevice here and below?

File: tools/ci-trigger/pr.go [hunk 2]
```diff
@@ -151,6 +158,41 @@ func (p *pullRequest) createBuild(ctx context.Context, buildClient *cloudbuild.S
 				}
 			}
 		}
+	physicalDeviceLoop:
+		for i, physicalDevice := range p.Physical {
+			if physicalDevice.Type == d {
+				if len(physicalDevice.Tests) == 0 {
+					continue
+				}
+				for _, v := range physicalDevice.Tests {
```
Comment (greg-dennis) [inline tools/ci-trigger/pr.go:167]: nit "t"->"v"?

File: tools/ci-trigger/pr.go [hunk 3]
```diff
@@ -151,6 +158,41 @@ func (p *pullRequest) createBuild(ctx context.Context, buildClient *cloudbuild.S
...
+				if err != nil {
+					return fmt.Errorf("json.Marshal device %q: %w", physicalDevice.Type.String(), err)
+				}
+				result := pubsubTopic.Publish(ctx, &pubsub.Message{Data: jsonMsg})
+				id, err := result.Get(ctx)
+				if err != nil {
+					return fmt.Errorf("pubsubTopic.Publish device %q: %w", physicalDevice.Type.String(), err)
+				}
+				glog.Infof("Sent Physical Test Job for PR%d at commit %q for device %q via %s", p.ID, p.HeadSHA, physicalDevice.Type.String(), id)
+				for j := range physicalDevice.Tests {
```
Comment (greg-dennis) [inline tools/ci-trigger/pr.go:191]: iterator over the values instead, i.e.

for _, t := range physicalDevice.Tests {
  t.Status = "setup"
}

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: tools/ci-trigger/pr.go (hunk 2 above)
Comment (bstoll) [reply tools/ci-trigger/pr.go]: Done

File: tools/ci-trigger/pr.go (hunk 3 above)
Comment (bstoll) [reply tools/ci-trigger/pr.go]: Done

File: tools/ci-trigger/pr.go (hunk 1 above)
Comment (bstoll) [reply tools/ci-trigger/pr.go]: It is but there were no pointers being used so modifying physicalDevice directly wouldn't work.  I updated this and think it is easier to follow now.

**Review by greg-dennis — APPROVED** [review]

**Pull request merged**